src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -4,8 +4,237 @@
+        public static System.Buffers.NIndex FromEnd(nint value) { throw null; }
+        public static System.Buffers.NIndex FromStart(nint value) { throw null; }
+        public override int GetHashCode() { throw null; }
+        public nint GetOffset(nint length) { throw null; }
+        public static explicit operator checked System.Index (System.Buffers.NIndex value) { throw null; }
+        public static explicit operator System.Index (System.Buffers.NIndex value) { throw null; }
+        public static implicit operator System.Buffers.NIndex (System.Index value) { throw null; }
+        public static implicit operator System.Buffers.NIndex (nint value) { throw null; }
+        public System.Index ToIndex() { throw null; }
+        public System.Index ToIndexUnchecked() { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public readonly partial struct NRange : System.IEquatable<System.Buffers.NRange>
+    {
+        private readonly int _dummyPrimitive;
+        public NRange(System.Buffers.NIndex start, System.Buffers.NIndex end) { throw null; }
+        public NRange(System.Range range) { throw null; }
+        public static System.Buffers.NRange All { get { throw null; } }
+        public System.Buffers.NIndex End { get { throw null; } }
+        public System.Buffers.NIndex Start { get { throw null; } }
+        public static System.Buffers.NRange EndAt(System.Buffers.NIndex end) { throw null; }
+        public bool Equals(System.Buffers.NRange other) { throw null; }
+        public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] object? value) { throw null; }
+        public override int GetHashCode() { throw null; }
+        public (nint Offset, nint Length) GetOffsetAndLength(nint length) { throw null; }
+        public static explicit operator checked System.Range (System.Buffers.NRange value) { throw null; }
+        public static explicit operator System.Range (System.Buffers.NRange value) { throw null; }
+        public static implicit operator System.Buffers.NRange (System.Range range) { throw null; }
+        public static System.Buffers.NRange StartAt(System.Buffers.NIndex start) { throw null; }
+        public System.Range ToRange() { throw null; }
+        public System.Range ToRangeUnchecked() { throw null; }
+        public override string ToString() { throw null; }
+    }
+}
 namespace System.Numerics.Tensors
 {
+    public partial interface ITensor<TSelf, T> : System.Collections.Generic.IEnumerable<T>, System.Collections.IEnumerable, System.IEquatable<TSelf>, System.Numerics.IEqualityOperators<TSelf, TSelf, bool> where TSelf : System.Numerics.Tensors.ITensor<TSelf, T>
+    {
+        static TSelf? Empty { get { throw null; } }
+        bool IsEmpty { get; }
+        bool IsPinned { get; }
+        ref T this[scoped System.ReadOnlySpan<nint> indices] { get; }
```

Comment:
The approved surface area has this returning `T` to avoid potential issues due to being an interface.

We're notably then missing:
```csharp
T this[params ReadOnlySpan<NIndex> indexes] { get; set; }
TSelf this[params scoped ReadOnlySpan<NRange> ranges] { get; set; }
```

Reply:
Added